File: binary_search/852. Peak Index in a Mountain Array/852.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        l = 0
        r = len(arr) - 1
        while ( r - l ) > 1:
            m = ( r + l ) // 2
            if arr[m] > arr[m+1]: r = m
            else: l = m
        
        return r

File: binary_search/852. Peak Index in a Mountain Array/852.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var peakIndexInMountainArray = function(arr) {
    let l = 0
    let r = len(arr) - 1
    while (( r - l ) > 1) {
        let m = Math.floor( r + l ) / 2
        if (arr[m] > arr[m+1])  r = m
        else l = m
    }
        
    return r
};